tools/ansible/tasks/wsu/test/e2e/wsu_test.go

```diff
@@ -288,3 +303,246 @@ func testHNSNetworksCreated(t *testing.T) {
+	err := pullDockerImage(windowsServerImage)
+	require.NoError(t, err, "Could not pull Windows Server image")
+
+	// This will run a Server on the container, which can be reached with a GET request
+	winServerCommand := []string{"powershell.exe", "-command",
+		"<#code used from https://gist.github.com/wagnerandrade/5424431#> ; $$listener = " +
+			"New-Object System.Net.HttpListener ; $$listener.Prefixes.Add('http://*:80/') ; " +
+			"$$listener.Start() ; $$callerCounts = @{} ; Write-Host('Listening at http://*:80/')" +
+			" ; while ($$listener.IsListening) { ;$$context = $$listener.GetContext() ;" +
+			"$$requestUrl = $$context.Request.Url ;$$clientIP = " +
+			"$$context.Request.RemoteEndPoint.Address ;$$response = $$context.Response ;" +
+			"Write-Host '' ;Write-Host('> {0}' -f $$requestUrl) ;  ;$$count = 1 ;" +
+			"$$k=$$callerCounts.Get_Item($$clientIP) ;if ($$k -ne $$null) { $$count += $$k } ;" +
+			"$$callerCounts.Set_Item($$clientIP, $$count) ;$$ip=(Get-NetAdapter | " +
+			"Get-NetIpAddress); $$header='<html><body><H1>Windows Container Web Server</H1>' " +
+			";$$callerCountsString='' ;$$callerCounts.Keys | % { $$callerCountsString+='<p>IP {0}" +
+			" callerCount {1} ' -f $$ip[1].IPAddress,$$callerCounts.Item($$_) } " +
+			";$$footer='</body></html>' ;$$content='{0}{1}{2}' -f " +
+			"$$header,$$callerCountsString,$$footer ;Write-Output $$content ;$$buffer = " +
+			"[System.Text.Encoding]::UTF8.GetBytes($$content) ;$$response.ContentLength64 = " +
+			"$$buffer.Length ;$$response.OutputStream.Write($$buffer, 0, $$buffer.Length) ;" +
+			"$$response.Close() ;$$responseStatus = $$response.StatusCode ;" +
+			"Write-Host('< {0}' -f $$responseStatus)  } ; "}
+	winServerDeployment, err := createWindowsServerDeployment("win-server", winServerCommand)
+	require.NoError(t, err, "Could not create Windows deployment")
+	defer deleteDeployment(winServerDeployment.Name)
+
+	// Wait until the server is ready to be queried
+	err = waitUntilDeploymentScaled(winServerDeployment.Name)
+	require.NoError(t, err, "Deployment was unable to scale")
+
+	// Get the pod so we can use its IP
+	winServerIP, err := getPodIP(*winServerDeployment.Spec.Selector)
+	require.NoError(t, err, "Could not retrieve pod with selector %v", *winServerDeployment.Spec.Selector)
+
+	// test Windows <-> Linux
+	// This will install curl and then curl the windows server.
+	linuxCurlerCommand := []string{"bash", "-c", "yum update; yum install curl -y; curl " + winServerIP}
+	linuxCurlerJob, err := createLinuxJob("linux-curler", linuxCurlerCommand)
+	require.NoError(t, err, "Could not create Linux job")
+	defer deleteJob(linuxCurlerJob.Name)
+	err = waitUntilJobSucceeds(linuxCurlerJob.Name)
+	assert.NoError(t, err, "Could not curl the Windows server from a linux container")
+
+	// test Windows <-> Windows on same node
+	// This will continually try to read from the Windows Server. We have to try multiple times as the network is flaky
```

Comment:
If there is a known bug around this call this out and explain that we're using loop to try multiple times

Reply:
Its not a known bug, its something that I've found when testing and may just be the normal behavior of Windows containers

Reply:
I don't think it is the network being flaky but us trying to run the tests before the container has finished its initial setup. That is what needs to be called out here.